iree/hal/vulkan/vulkan_device.cc

```diff
@@ -714,6 +788,34 @@ Status VulkanDevice::WaitSemaphores(absl::Span<const SemaphoreValue> semaphores,
   // semaphores we waited on (including those already expired above).
 
   return OkStatus();
+
+#endif  // IREE_HAL_VULKAN_EMULATE_TIMELINE_SEMAPHORES
+}
+
+Status VulkanDevice::OnSemaphoreSignal(Semaphore* /*semaphore*/) {
+#if IREE_HAL_VULKAN_EMULATE_TIMELINE_SEMAPHORES
+  for (CommandQueue* queue : dispatch_queues_) {
```

Comment:
it'd be worth having an all_queues_ list that tracks each unique queue and iterating over that instead - the dispatch/transfer_queues_ lists may contain duplicates, and that'd be confusing and create redundant work if the same queue is advanced twice for a single signal

Reply:
Ah, actually we have `command_queues_` for that. I missed it previously.